libs/ml/include/ml/dataloaders/word2vec_loaders/sgns_w2v_dataloader.hpp

```diff
@@ -396,7 +395,7 @@ void GraphW2VLoader<T>::BufferNextSamples()
             window_size_;  // the current word start from position that allows full context window
         current_sentence_++;
       }
-      if (IsDone())
+      if (IsDone() || current_sentence_ >= data_.size())
```

Comment:
If current_sentence_ >= data_.size() won't IsDone() be true anyway? I think the second condition here is unnecessary.

Reply:
IsDone is checking for data in buffer and when there are data it continues, but we can be in next non-existing sentence and crash. This was the only way I found to pass W2V tests and not be crashing on distributed W2V training.